src/Symfony/Component/HttpClient/HttplugClient.php

```diff
@@ -49,25 +55,65 @@
  *
  * @author Nicolas Grekas <p@tchwork.com>
  */
-final class HttplugClient implements HttpClient, RequestFactory, StreamFactory, UriFactory
+final class HttplugClient implements HttpClient, RequestFactory, StreamFactory, UriFactory, HttpAsyncClient
 {
     private $client;
+    private $responseFactory;
+    private $streamFactory;
 
     public function __construct(HttpClientInterface $client = null, ResponseFactoryInterface $responseFactory = null, StreamFactoryInterface $streamFactory = null)
     {
-        $this->client = new Psr18Client($client, $responseFactory, $streamFactory);
+        $this->client = $client ?? \Symfony\Component\HttpClient\HttpClient::create();
```

Comment:
Should I import conflicting class/interface names or should I leave it like this?

Reply:
I'd say `HttpClient` should be aliased to e.g. `HttplugClientInterface`?